pkg/utils/utils.go

```diff
@@ -311,3 +311,101 @@ func Remove(arr []string, item string) []string {
 	}
 	return arr
 }
+
+var GlobalK8NSObj *K8ValidNamespaces = &K8ValidNamespaces{}
+
+func GetGlobalK8NSObj() *K8ValidNamespaces {
+	return GlobalK8NSObj
+}
+
+func IsNSPresent(namespace string, obj *K8ValidNamespaces) bool {
+	obj.validNSList.lock.RLock()
+	defer obj.validNSList.lock.RUnlock()
+	_, flag := obj.validNSList.nsList[namespace]
+	AviLog.Debugf("Is Namespace [%s] valid? -->[%v]", namespace, flag)
+	return flag
+}
+
+func InitializeNSSync(label string) {
+	AviLog.Debugf("In InitializeNSSync function, namespace label to check is: %s\n", label)
+	if label != "" {
+		segments := strings.Split(label, "=")
+		seg0 := strings.Trim(segments[0], " ")
+		seg1 := strings.Trim(segments[1], " ")
+		if len(segments) != 2 || seg0 == "" || seg1 == "" {
+			AviLog.Error("Error occured while parsing labels.")
+			return
+		}
+		GlobalK8NSObj.EnableMigration = true
+		GlobalK8NSObj.nsFilter.key = seg0
+		GlobalK8NSObj.nsFilter.value = seg1
+		GlobalK8NSObj.validNSList.nsList = make(map[string]bool)
+	}
+}
+
+//Get namespace label filter key and value
+func GetNSFilter(obj *K8ValidNamespaces) (string, string) {
+	var key string
+	var value string
+	if obj.nsFilter.key != "" {
+		key = obj.nsFilter.key
+	}
+	if obj.nsFilter.value != "" {
+		value = obj.nsFilter.value
+	}
+	return key, value
+}
+
+func AddNamespace(namespace string, obj *K8ValidNamespaces) {
+	obj.validNSList.lock.Lock()
```

Comment:
Over nsList ... reading and writing is going on from different points. like add, update namespace event handler, reading from BuildL7VSGraph etc... I think this will be in two different threads.. so added lock before writing.

Reply:
Okay if you are just adding things from one thread, maybe you just need a READ lock. You currently have a R/W lock.

Reply:
As we are updating nsList here, we added write lock here.